osquery/tables/networking/linux/process_open_sockets.cpp

```diff
@@ -27,12 +27,16 @@ QueryData genOpenSockets(QueryContext& context) {
    * otherwise query all pids from the system and also report on sockets without
    * an associated pid.
    */
-  bool pid_filter = false;
+  bool pid_filter;
```

Comment:
Is this needed here? Maybe down below you could just have
```
auto pid_filter = !pids.empty() && std::find(pids.begin(), pids.end(), "-1") != pids.end();
```

Reply:
Sure that makes sense.